pandas/tests/extension/masked_shared.py

```diff
@@ -64,6 +64,37 @@ def check_reduce(self, ser: pd.Series, op_name: str, skipna: bool):
                 expected = pd.NA
         tm.assert_almost_equal(result, expected)
 
+    def check_reduce_and_wrap(self, ser: pd.Series, op_name: str, skipna: bool):
+        if op_name in ["count", "kurt", "sem"]:
+            pytest.skip(f"{op_name} not an array method")
+
+        arr = ser.array
+
+        is_windows_or_32bit = is_platform_windows() or not IS64
+
+        if tm.is_float_dtype(arr.dtype):
+            cmp_dtype = arr.dtype.name
+        elif op_name in ["mean", "median", "var", "std", "skew"]:
+            cmp_dtype = "Float64"
+        elif op_name in ["max", "min"]:
+            cmp_dtype = arr.dtype.name
+        elif arr.dtype in ["Int64", "UInt64"]:
+            cmp_dtype = arr.dtype.name
+        elif tm.is_signed_integer_dtype(arr.dtype):
+            cmp_dtype = "Int32" if is_windows_or_32bit else "Int64"
+        elif tm.is_unsigned_integer_dtype(arr.dtype):
+            cmp_dtype = "UInt32" if is_windows_or_32bit else "UInt64"
+        else:
+            raise TypeError("not supposed to reach this")
+
+        result = arr._reduce_and_wrap(op_name, skipna=skipna, kwargs={})
```

Comment:
Instead of calling this directly, should we test with `result = DataFrame({"col": arr}).op(...).array` instead?

Reply:
Yes, that would be good too. I've added it, but I think because `__reduce_and_wrap` is part of the `ExtensionArray` interface, it't be good to test that also, so I test both in the updated version.